src/main/java/seedu/address/storage/JsonSerializableLeavesBook.java

```diff
@@ -0,0 +1,79 @@
+package seedu.address.storage;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonRootName;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.AddressBook;
+import seedu.address.model.LeavesBook;
+import seedu.address.model.ReadOnlyLeavesBook;
+import seedu.address.model.leave.Date;
+import seedu.address.model.leave.Leave;
+import seedu.address.model.leave.Status;
+import seedu.address.model.person.Person;
+
+/**
+ * An Immutable LeavesBook that is serializable to JSON format.
+ */
+@JsonRootName(value = "leavesbook")
+public class JsonSerializableLeavesBook {
+
+    public static final String MESSAGE_DUPLICATE_LEAVE = "Leaves list contains duplicate leave(s).";
+    private final List<JsonAdaptedLeave> leaves = new ArrayList<>();
+
+    /**
+     * Constructs a {@code JsonSerializableLeavesBook} with the given leaves.
+     */
+    @JsonCreator
+    public JsonSerializableLeavesBook(@JsonProperty("leaves") List<JsonAdaptedLeave> leaves) {
+        this.leaves.addAll(leaves);
+    }
+
+    /**
+     * Converts a given {@code ReadOnlyLeavesBook} into this class for Jackson use.
+     *
+     * @param source future changes to this will not affect the created {@code JsonSerializableLeavesBook}.
+     */
+    public JsonSerializableLeavesBook(ReadOnlyLeavesBook source) {
+        leaves.addAll(source.getLeaveList().stream().map(JsonAdaptedLeave::new).collect(Collectors.toList()));
+    }
+
+    /**
+     * Converts this leaves book into the model's {@code LeavesBook} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated.
+     */
+    // public LeavesBook toModelType(AddressBook addressBook) throws IllegalValueException {
+    //     LeavesBook leavesBook = new LeavesBook();
+    //     for (JsonAdaptedLeave jsonAdaptedLeave : leaves) {
+    //         Leave leave = jsonAdaptedLeave.toModelType();
+    //         if (leavesBook.hasLeave(leave)) {
+    //             throw new IllegalValueException(MESSAGE_DUPLICATE_LEAVE);
+    //         }
+    //         leavesBook.addLeave(leave);
+    //     }
+    //     return leavesBook;
+    // }
+    public LeavesBook toModelType(AddressBook addressBook) throws IllegalValueException {
+        LeavesBook leavesBook = new LeavesBook();
+        for (JsonAdaptedLeave jsonAdaptedLeave : leaves) {
+            String title = jsonAdaptedLeave.getTitle();
```

Comment:
Would it be better to just construct the Leave by calling JsonAdaptedLeave::toModelType()?

Reply:
@ivyy-poison 
Re: toModelType() alternative

If we remove the toModelType() method, how would the Leave be constructed, and who is responsible for its construction? Also I'm not really sure I understand the part for JsonSerializableLeavesBook and its justification

Reply:
JsonAdaptedLeave::toModelType() returns a Leaves object from Json format. (If we first assume without using your Dummy Person solution) We are unable to create the Person object required for a Leaves object without passing in the addressBook here as well, which would turn the signature into JsonAdaptedLeave::toModelType(AddressBook), which I feel violates the separation of concern and the various business layers. My current solution is to expose all the fields of the JsonAdaptedLeave object, ie getEmployee(), getStart(), getEnd(), .... which will then be used in the layer above: JsonSerializableLeavesBook::toModelType(AddressBook), where with the addressBook it can now search for the corresponding person, if present, and generate the Leaves Object required to. be put into the LeavesBook. Essentially I shifted this responsibility up one level. I would think it is more appropriate to have the JsonSerializableLeavesBook handle the AddressBook, instead of passing the addressbook down to the adaptedLeaves level and let it use it to generate the Leaves object in the toModelType() method instead

Reply:
I'm not sure I understand why the addressBook needs to be passed down in the first place. Since JsonAdaptedLeave already contains a string containing the user's name, we can just create the Person field in the Leave via Person(name, null, null...) or equivalent. So long as we use isSamePerson when comparing between the Person in the Leave instance and the Person in the addressBook instance, they are functionally the same person. Additionally, I'm not sure if there is an advantage in maintaining referential integrity - whenever the employee contact is modified/deleted, the Person instance in the addressBook is completely replaced/deleted, thus losing referential integrity.

Reply:
Got it, thank you